### PR TITLE
Add when to update operator doc

### PR DESCRIPTION
**Pull request by anik120**

### CODE REVIEW

File: docs/when-to-update-my-operator.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+```
+$ oc get catsrc operatorhubio-catalog -n olm -o yaml | grep image:
+    
+    image: quay.io/operator-framework/upstream-community-operators:0.0.1
+
+$ kubectl patch catsrc operatorhubio-catalog -n olm --type=merge -p '{"spec": {"image": "quay.io/operator-framework/upstream-community-operators:0.0.2"}}'
+
+```
+
+If the image used to build the `Catalogsource` uses the `latest` tag, simply delete the pod corresponding to the `CatalogSource`. When the pod is recreated, it will be recreated with the latest image of the catalog, which will contain updates to the operators in that catalog.
```
Comment (Bowenislandsong) [inline docs/when-to-update-my-operator.md]: Maybe we can also mention checking the pullimage tag is set to `always` here?

File: docs/when-to-update-my-operator.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+For example:
+
+```
+$ kubectl get pods -n olm
+
+    NAME                                   READY   STATUS    RESTARTS   AGE
+    operatorhubio-catalog-xl4rg            0/1     Running   0          12h
+
+$ kubectl delete pod operatorhubio-catalog-xl4rg -n olm
```
Comment (Bowenislandsong) [inline docs/when-to-update-my-operator.md]: maybe you can change this to a oneliner?
```bash
$ kubectl delete pods -n olm -l olm.catalogSource=operatorhubio-catalog
```
This way we delete by label.

File: docs/when-to-update-my-operator.md (hunk 1 above)
Comment (anik120) [reply docs/when-to-update-my-operator.md]: @Bowenislandsong we omit setting the `imagePullPolicy` explicitly [here](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/registry/reconciler/reconciler.go#L89), so with the `latest` tag, the `imagePullPolicy`is implicitly set to `Always`. You can refer to [this doc](https://kubernetes.io/docs/concepts/containers/images/#updating-images) for more info.

File: docs/when-to-update-my-operator.md (hunk 2 above)
Comment (anik120) [reply docs/when-to-update-my-operator.md]: Good point.

File: docs/when-to-update-my-operator.md (hunk 1 above)
Comment (Bowenislandsong) [reply docs/when-to-update-my-operator.md]: Oh nice.

File: docs/when-to-update-my-operator.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# When do I need to update my operator
```
Comment (awgreene) [inline docs/when-to-update-my-operator.md]: @dmesser it is unclear to me if this was meant to be written for a `cluster administrator` or an `operator author` - could you add context?

Comment (awgreene) [reply docs/when-to-update-my-operator.md]: As a developer - I would look through the `OLM-BOOK` contents and assume that this document tells me when to update an API.

**Pull request merged**